worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java

```diff
@@ -52,8 +55,41 @@ public CoverageDriverImpl(ContractService contractService, CoverageService cover
      * and coverage information that is too old.
      */
     @Override
-    public void queueStaleCoveragePeriods() {
+    public void queueStaleCoveragePeriods() throws InterruptedException {
 
+        Lock lock = coverageLockWrapper.getCoverageLock();
+        boolean locked = false;
+
+        try {
+
+            Set<CoveragePeriod> outOfDateInfo = getCoveragePeriods();
+
+            log.info("queueing all stale coverage periods");
+
+            // Job runs once a day so we need to grab this lock
+            locked = lock.tryLock(10 * SIXTY_SECONDS, TimeUnit.SECONDS);
```

Comment:
`lock.tryLock(10, TimeUnit.MINUTES);`

is a lot more direct and clearer to the reader.